### PR TITLE
Serving React (w/ React Router & TypeScript) using Fiber

### PR DESCRIPTION
**Pull request by lauslim12**
## Introduction

Hello!

First off, thanks for your hard work of creating and maintaining this framework. It helps a lot! :)

To contribute to the Fiber project, I have this pull request to to be integrated into one of Fiber's examples. This application showcases how to serve static SPAs that are a bit complex (in this case, using React, TypeScript, and React Router) using the Fiber framework. I know some people have had some problems with that (especially in the routing part and how to serve the frontend part of the application properly), and with this sample application, I hope they will have an idea about how to do exactly that.

Another way to serve the application is by using nginx to serve the frontend, and also using nginx to serve the backend if the request reaches a certain part (`/api` for example), but it will not be covered in this example.

This application has three paths: `/`, `/react`, and a catch-all, 404 route. I have also ensured that the routing will be available even if you manually type the URL into the browser (React apps are famous for that client-side routing). This application is named as `react-fiber`, not to be confused with Fiber algorithm that React uses.

This application comes with full documentation, and a Docker script (multi-stage for memory considerations) as a reference and to run this application with minimum configurations. Just `docker run` will do.

This PR also re-sorts the examples list in the main `README.md` file for easier searching.

## Images

Several screenshots of the application:

### `/` Route:

react-fiber-fiber

### `/react` Route:

react-fiber-react

### 404 Route:

react-fiber-404

### Responsive Design:

react-fiber-responsive

## Closing

Thanks for your attention and hard work! :)

### CODE REVIEW

File: react-fiber/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+- Go with Fiber
+- React with TypeScript and React Router
+- Docker
+
+## Application
+
+- This application has three routes: `/`, `/react`, and a catch-all, 404 route. `/` will show the Fiber logo, `/react` will show the React logo, and the 404 route will show both logos.
+- As this application serves the frontend while backed by a server, the client-side routing will work well and will not cause any issue (unlike if you are running without a file server). You can type the URL route manually in the browser and it will still work and will render the accurate page, so no worries.
+- This is a simplified form of Create React App with TypeScript. With that being said, that's why there is no `manifest.json`, `logo512.png`, and other stuff like that.
```
Comment (hi019) [inline react-fiber/README.md:15]:
```suggestion
- This is a simplified form of Create React App with TypeScript. With that being said, that's why there is no `manifest.json`, `logo512.png`, and other extra things like that.
```

File: README.md [hunk 2]
```diff
@@ -15,32 +15,33 @@ Here you can find the most **delicious** recipes to cook delicious meals using o
...
+- [JWT](/jwt)
 - [Kubernetes](/k8s)
 - [Listen on Multiple Ports](/multiple-ports)
 - [Live Reloading (Air)](/air)
 - [MongoDB](/mongodb)
 - [Netlify Functions](fiber-svelte-netlify)
-- [RabbitMQ](rabbitmq)
 - [PostgreSQL](/postgresql)
+- [RabbitMQ](rabbitmq)
+- [React Fiber](/react-fiber)
```
Comment (hi019) [inline README.md:34]:
```suggestion
- [React Router](/react-fiber)
```

**Comment by hi019** [comment]
Thanks for the PR! Can you rename it to be `react-router`?

**Comment by lauslim12** [comment]
Sure thing! Just a minute.

**Comment by lauslim12** [comment]
Fixed!

**Review by ReneWerner87 — APPROVED** [review]
Thx for the contribution

**Pull request merged**